src/components/PlayerControlls.jsx

```diff
@@ -155,26 +67,227 @@ const GeneralButtons = ({ state, setState, battle, setBattle, levelManager, setL
+      } else {
+        console.log('No Monsters Left');
+      }
+    }
+    if (e.key === '8' && e.ctrlKey) {
+      e.preventDefault();
+      if (state.playerHand[7]) {
+        setBattle({ ...battle, selectedCard: state.playerHand[7] });
+      } else {
+        console.log('No Cards in hand');
+      }
+    }
+    if (e.key === '9' && e.altKey) {
+      e.preventDefault();
+      if (levelManager.wave[8]) {
+        setBattle({ ...battle, targetMonster: levelManager.wave[8] });
+      } else {
+        console.log('No Monsters Left');
+      }
+    }
+    if (e.key === '9' && e.ctrlKey) {
+      e.preventDefault();
+      if (state.playerHand[8]) {
+        setBattle({ ...battle, selectedCard: state.playerHand[8] });
+      } else {
+        console.log('No Cards in hand');
+      }
+    }
+    if (e.key === '0' && e.altKey) {
+      e.preventDefault();
+      if (levelManager.wave[9]) {
+        setBattle({ ...battle, targetMonster: levelManager.wave[9] });
+      } else {
+        console.log('No Monsters Left');
+      }
+    }
+    if (e.key === '0' && e.ctrlKey) {
+      e.preventDefault();
+      setBattle({ ...battle, selectedCard: null });
+    }
+    if (e.key === 'h') {
+      console.log('h pressed');
+      battle.selectedCard.effect();
+    }
     if (e.key === 'a') {
-      state.controller.attack(
-        battle.targetHero,
-        battle.targetMonster,
-        levelManager.wave,
-      );
+      const mana = state.controller.mana;
+      if (battle.selectedCard && battle.selectedCard.manaCost) {
+        const manaCost = battle.selectedCard.manaCost;
+        if (mana >= manaCost) {
+          battle.selectedCard.effect(
+            battle,
+            setBattle,
+            battle.targetMonster,
+            state.computer,
+            levelManager.wave
+          );
+
+          let manaDrain = battle.selectedCard.manaCost;
+          let currentMana = state.controller.mana - manaDrain;
+          setState({
+            ...state,
+            controller: {
+              ...state.controller,
+              mana: currentMana,
+            },
+          });
+        } else {
+          let message = 'Not enough mana';
+          console.log('Not enough mana');
+          battle.heroDamageAnimation = message;
+          setBattle({
+            ...battle,
+            heroDamageAnimation: battle.heroDamageAnimation,
+          });
+        }
+      } else {
+        state.controller.attack(
+          battle.targetHero,
+          battle.targetMonster,
+          levelManager.wave
+        );
+        const damage = battle.targetHero.attackPower;
+        console.log(damage);
+        battle.heroDamageAnimation = damage;
+        setBattle({
+          ...battle,
+          heroDamageAnimation: damage,
+        });
+      }
+      setTimeout(() => {
+        setBattle({
+          ...battle,
+          heroDamageAnimation: null,
+        });
+      }, 1000);
```

Comment:
I get what is done here, 
It seems to me this could be more DRY for sure, I don't immediately know how though.